src/model/HeatExchangerDesiccantBalancedFlow.cpp

```diff
@@ -0,0 +1,346 @@
+        OptionalSchedule intermediate = modelObject->optionalCast<Schedule>();
+        if (intermediate) {
+          Schedule schedule(*intermediate);
+          return setAvailabilitySchedule(schedule);
+        }
+      }
+      return false;
+    }
+
+    unsigned HeatExchangerDesiccantBalancedFlow_Impl::primaryAirInletPort() const {
+      return OS_HeatExchanger_Desiccant_BalancedFlowFields::RegenerationAirInletNode;
+    }
+
+    unsigned HeatExchangerDesiccantBalancedFlow_Impl::primaryAirOutletPort() const {
+      return OS_HeatExchanger_Desiccant_BalancedFlowFields::RegenerationAirOutletNode;
+    }
+
+    unsigned HeatExchangerDesiccantBalancedFlow_Impl::secondaryAirInletPort() const {
+      return OS_HeatExchanger_Desiccant_BalancedFlowFields::ProcessAirInletNode;
+    }
+
+    unsigned HeatExchangerDesiccantBalancedFlow_Impl::secondaryAirOutletPort() const {
+      return OS_HeatExchanger_Desiccant_BalancedFlowFields::ProcessAirOutletNode;
+    }
+
+    AirflowNetworkEquivalentDuct HeatExchangerDesiccantBalancedFlow_Impl::getAirflowNetworkEquivalentDuct(double length, double diameter) {
+      boost::optional<AirflowNetworkEquivalentDuct> opt = airflowNetworkEquivalentDuct();
+      if (opt) {
+        if (opt->airPathLength() != length) {
+          opt->setAirPathLength(length);
+        }
+        if (opt->airPathHydraulicDiameter() != diameter) {
+          opt->setAirPathHydraulicDiameter(diameter);
+        }
+      }
+      return AirflowNetworkEquivalentDuct(model(), length, diameter, handle());
+    }
+
+    boost::optional<AirflowNetworkEquivalentDuct> HeatExchangerDesiccantBalancedFlow_Impl::airflowNetworkEquivalentDuct() const {
+      std::vector<AirflowNetworkEquivalentDuct> myAFN =
+        getObject<ModelObject>().getModelObjectSources<AirflowNetworkEquivalentDuct>(AirflowNetworkEquivalentDuct::iddObjectType());
+      auto count = myAFN.size();
+      if (count == 1) {
+        return myAFN[0];
+      } else if (count > 1) {
+        LOG(Warn, briefDescription() << " has more than one AirflowNetwork EquivalentDuct attached, returning first.");
+        return myAFN[0];
+      }
+      return boost::none;
+    }
+
+  }  // namespace detail
+
+  HeatExchangerDesiccantBalancedFlow::HeatExchangerDesiccantBalancedFlow(const Model& model)
+    : AirToAirComponent(HeatExchangerDesiccantBalancedFlow::iddObjectType(), model) {
+    OS_ASSERT(getImpl<detail::HeatExchangerDesiccantBalancedFlow_Impl>());
+
+    Schedule schedule = model.alwaysOnDiscreteSchedule();
+    setAvailabilitySchedule(schedule);
+
+    HeatExchangerDesiccantBalancedFlowPerformanceDataType1 heatExchangerPerformance(model);
+    bool ok = setHeatExchangerPerformance(heatExchangerPerformance);
+    if (!ok) {
+      remove();
+      LOG_AND_THROW("Unable to set " << briefDescription() << "'s heatExchangerDesiccantBalancedFlowPerformanceDataType1 to "
+                                     << heatExchangerPerformance.briefDescription() << ".");
+    }
```

Comment:
here since you are instantiating the object yourself and KNOW it should work, you can just do this:

```suggestion
    OS_ASSERT(ok);
```

A debug build would catch a new problem, but the release build wouldn't get affected by the extra check.